service/framework/kvmconfig/framework.go

```diff
@@ -0,0 +1,182 @@
+package kvmconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/kvm-operator/service/framework/kvmconfigv2"
+	"github.com/giantswarm/kvm-operator/service/framework/kvmconfigv3"
+	"github.com/giantswarm/kvm-operator/service/keyv3"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/context/updateallowedcontext"
+	"github.com/giantswarm/operatorkit/informer"
+	"github.com/giantswarm/randomkeys"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+)
+
+type FrameworkConfig struct {
+	G8sClient    versioned.Interface
+	K8sClient    kubernetes.Interface
+	K8sExtClient apiextensionsclient.Interface
+	Logger       micrologger.Logger
+
+	GuestUpdateEnabled bool
+	// Name is the name of the project.
+	Name string
+}
```

Comment:
As we said no `Default*Config`, no comments anymore. I'll have to update operatorkit to cleanup the rest.